genomesubsampler/prototypeSelection.py

```diff
@@ -41,6 +41,56 @@
 import scipy as sp
 
 
+def _validate_parameters(dm, num_prototypes, seedset=[]):
+    '''Validate the paramters for each algorithm.
+
+    Parameters
+    ----------
+    dm: skbio.stats.distance.DistanceMatrix
+        Pairwise distances for all elements in the full set S.
+    num_prototypes: int
+        Number of prototypes to select for distance matrix.
+        Must be >= 2, since a single prototype is useless.
+        Must be smaller than the number of elements in the distance matrix,
+        otherwise no reduction is necessary.
+    seedset: iterable of str
+        A set of element IDs that are preferably selected for the resulting
+        set. Must be smaller than the number of prototypes to select. All get
+        selected first, after which the algorithms will operate based on them.
+        Warning: It will most likely violate the global objective function to
+        pre-select elements.
+
+    Raises
+    ------
+    ValueError
+        The number of prototypes to be found should be at least 2 and at most
+        one element smaller than elements in the distance matrix. Otherwise, a
+        ValueError is raised.
+        The IDs in the seed set must be unique, and must be present in the
+        distance matrix. Otherwise, a ValueError is raised.
+        The size of the seed set must be smaller than the number of prototypes
+        to be found. Otherwise, a ValueError is raised.
+    '''
+    if num_prototypes < 2:
+        raise ValueError("'num_prototypes' must be >= 2, since a single "
+                         "prototype is useless.")
+    if num_prototypes >= dm.shape[0]:
+        raise ValueError("'num_prototypes' must be smaller than the number of "
+                         "elements in the distance matrix, otherwise no "
+                         "reduction is necessary.")
+    if len(seedset) > 0:
+        seeds = set(seedset)
+        if len(seeds) < len(seedset):
+            raise ValueError("There are duplicated IDs in 'seedset'.")
+        if not seeds < set(dm.ids):
+            raise ValueError("'seedset' is not a subset of the element IDs in "
+                             "the distance matrix.")
+        if len(seeds) >= num_prototypes:
+            raise ValueError("Size of 'seedset' must be smaller than the "
+                             "number of prototypes to select, otherwise no "
+                             "selection is necessary.")
```

Comment:
I think this error message doesn't have to be that complicated. Saying "Size of 'seedset' must be smaller than the number of prototypes to select." should be enough. The user just needs to know the rule, but doesn't have to understand the mechanism, from a simple error message.